python/tests/test_extracted_datatypes.py

```diff
@@ -48,39 +80,95 @@ def sample_input_d(ceiling_input: reader.InputData,
         ],
         'floors': [
             floor_input
-        ]
+        ],
+        'walls': [
+            wall_input
+        ],
+        'codes': codes
+
     }
 
 
 class TestCeiling:
 
     @pytest.fixture
-    def sample(self, sample_input_d: reader.InputData):
+    def sample(self, sample_input_d: reader.InputData) -> typing.Dict[str, str]:
         ceiling = sample_input_d['ceilings'][0]
         ceiling['nominalRsi'] = float(ceiling['nominalRsi'])
         ceiling['effectiveRsi'] = float(ceiling['effectiveRsi'])
         ceiling['area'] = float(ceiling['area'])
         ceiling['length'] = float(ceiling['length'])
         return ceiling
 
-    def test_from_data(self, sample):
+    def test_from_data(self, sample: typing.Dict[str, typing.Any]) -> None:
         output = extracted_datatypes.Ceiling.from_data(sample)
         assert output.label == 'Main attic'
         assert output.area_metres == 46.4515
 
+    def test_to_dict(self, sample: typing.Dict[str, typing.Any]) -> None:
+        output = extracted_datatypes.Ceiling.from_data(sample).to_dict()
+        assert output['areaMetres'] == sample['area']
+
 
 class TestFloor:
 
     @pytest.fixture
-    def sample(self, sample_input_d: reader.InputData):
+    def sample(self, sample_input_d: reader.InputData) -> typing.Dict[str, typing.Any]:
         floor = sample_input_d['floors'][0]
         floor['nominalRsi'] = float(floor['nominalRsi'])
         floor['effectiveRsi'] = float(floor['effectiveRsi'])
         floor['area'] = float(floor['area'])
         floor['length'] = float(floor['length'])
         return floor
 
-    def test_from_data(self, sample):
+    def test_from_data(self, sample: typing.Dict[str, typing.Any]) -> None:
         output = extracted_datatypes.Floor.from_data(sample)
         assert output.label == 'Rm over garage'
         assert output.area_metres == 9.2903
+
+    def test_to_dict(self, sample: typing.Dict[str, typing.Any]) -> None:
+        output = extracted_datatypes.Floor.from_data(sample).to_dict()
+        assert output['areaMetres'] == sample['area']
+
+
+def _dict_codes(codes: typing.Dict[str, typing.List[typing.Dict[str, str]]]
```

Comment:
This duplicates private functionality from `dwelling.py`. This is usually an anti-pattern. Either the private function should be made public, or (better) the tests are modified to not need it.